test/parallel/test-net-better-error-messages-path.js

```diff
@@ -2,12 +2,20 @@
 const common = require('../common');
 const net = require('net');
 const assert = require('assert');
-const fp = '/tmp/fadagagsdfgsdf';
-const c = net.connect(fp);
 
-c.on('connect', common.mustNotCall());
+{
+  const fp = '/tmp/fadagagsdfgsdf';
+  const c = net.connect(fp);
 
-c.on('error', common.mustCall(function(e) {
-  assert.strictEqual(e.code, 'ENOENT');
-  assert.strictEqual(e.message, `connect ENOENT ${fp}`);
-}));
+  c.on('connect', common.mustNotCall());
+  c.on('error', common.mustCall(function(e) {
+    assert.strictEqual(e.code, 'ENOENT');
+    assert.strictEqual(e.message, `connect ENOENT ${fp}`);
+  }));
+}
+
+{
+  assert.throws(() => {
+    net.createConnection({ path: {} });
+  }, TypeError);
```

Comment:
Re: asserting error messages, Could you use [`expectsError`](https://github.com/nodejs/node/tree/master/test/common#expectserrorsettings)